### PR TITLE
Full implementation

### PR DESCRIPTION
**Pull request by FrSanchez**
Add proper cleanup
Fix insert
Add retrieve

### CODE REVIEW

**Comment by FrSanchez** [comment]
Most memory errors I could found where solved
displaySideways implemented - somewhat different from example
getHeight implemented
Changed makeEmpty to fix issues with cleanup
Added initialization to pointers to avoid access errors
Added check for instantiation with copy to avoid copying the same element

**Comment by FrSanchez** [comment]
@dianagut make sure you merge the my change that includes a working version of the code

File: BinTree/BinTree.h [hunk 1]
```diff
@@ -19,18 +19,30 @@ class BinTree {
 public:
     BinTree();
     ~BinTree();
-    int getHeight();
-    bool insertNode(NodeData*);
+    BinTree(const BinTree& bt);
+    
+    int getHeight() const;
+    int getHeight (const NodeData &) const;
```
Comment (FrSanchez) [inline BinTree/BinTree.h:25]: getHeigh(NodeData&) is the method requested in the PDF. The other is not required.

File: BinTree/BinTree.cpp [hunk 2]
```diff
@@ -84,6 +213,57 @@ std::ostream& operator<<(std::ostream& os, const BinTree& p)
...
+    }
+    if (left) {
+        left->displaySideways(spacer);
+    }
+}
+
+int BinTree::getHeight(const NodeData &node) const
+{
+    if (*root == node) {
+        return getHeight();
```
Comment (FrSanchez) [inline BinTree/BinTree.cpp:253]: When finding the requested node, getHeight() will let us know how far this node is from the leaf (or if this is the leaf).

**Review by dianagut — APPROVED** [review]

**Review by dianagut — APPROVED** [review]

**Pull request merged**